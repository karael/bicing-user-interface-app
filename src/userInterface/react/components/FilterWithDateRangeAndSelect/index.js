import React from 'react';
import DatePicker from 'react-datepicker';
import Select from 'react-select';
import moment from 'moment';

import 'userInterface/react/components/FilterWithDateRangeAndSelect/stylesheets/react-datepicker.css';

const momentFormat = 'YYYY-MM-DD HH:mm:ss';

const filterWithDateRangeAndSelect = ({
    dateStart,
    dateEnd,
    option,
    options,
    onDateStartChange,
    onDateEndChange,
    onOptionChange
}) => {
    return <div>
        <DatePicker
            selected={moment(dateStart, momentFormat)}
            selectsStart
            startDate={moment(dateStart, momentFormat)}
            endDate={moment(dateEnd, momentFormat)}
            onChange={(date) => onDateStartChange(date.format(momentFormat))}
            showTimeSelect
            timeFormat="HH:mm"
            timeIntervals={5}
            dateFormat="LLL"
            timeCaption="time"
        />
        <DatePicker
            selected={moment(dateEnd, momentFormat)}
            selectsEnd
            startDate={moment(dateStart, momentFormat)}
            endDate={moment(dateEnd, momentFormat)}
            onChange={(date) => onDateEndChange(date.format(momentFormat))}
            showTimeSelect
            timeFormat="HH:mm"
            timeIntervals={5}
            dateFormat="LLL"
            timeCaption="time"
        />
        <Select
            value={options.find(optionObj => optionObj.value === option)}
            onChange={(selectedOption) => onOptionChange(selectedOption.value)}
            options={options}
        />
    </div>
};

export default filterWithDateRangeAndSelect;
