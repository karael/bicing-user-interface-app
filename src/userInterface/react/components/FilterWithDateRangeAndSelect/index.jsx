import {
  arrayOf, func, shape, string,
} from 'prop-types';
import moment from 'moment';
import React from 'react';
import DatePicker from 'react-datepicker';
import Select from 'react-select';

import 'userInterface/react/components/FilterWithDateRangeAndSelect/stylesheets/react-datepicker.css';

// @todo clean and separate component
const filterWithDateRangeAndSelect = ({
  dateStart,
  dateEnd,
  option,
  options,
  onDateStartChange,
  onDateEndChange,
  onOptionChange,
  momentFormat,
}) => (
  <div>
    <DatePicker
      selected={moment(dateStart, momentFormat)}
      selectsStart
      startDate={moment(dateStart, momentFormat)}
      endDate={moment(dateEnd, momentFormat)}
      onChange={date => onDateStartChange(date.format(momentFormat))}
      showTimeSelect
      timeFormat="HH:mm"
      timeIntervals={5}
      dateFormat="LLL"
      timeCaption="time"
    />
    <DatePicker
      selected={moment(dateEnd, momentFormat)}
      selectsEnd
      startDate={moment(dateStart, momentFormat)}
      endDate={moment(dateEnd, momentFormat)}
      onChange={date => onDateEndChange(date.format(momentFormat))}
      showTimeSelect
      timeFormat="HH:mm"
      timeIntervals={5}
      dateFormat="LLL"
      timeCaption="time"
    />
    <Select
      value={options.find(optionObj => optionObj.value === option)}
      onChange={selectedOption => onOptionChange(selectedOption.value)}
      options={options}
    />
  </div>
);

filterWithDateRangeAndSelect.propTypes = {
  dateStart: string,
  dateEnd: string,
  option: string,
  options: arrayOf(shape({
    value: string.isRequired,
    label: string.isRequired,
  })),
  onDateStartChange: func.isRequired,
  onDateEndChange: func.isRequired,
  onOptionChange: func.isRequired,
  momentFormat: string,
};

filterWithDateRangeAndSelect.defaultProps = {
  dateStart: moment().subtract(1, 'hour').format('YYYY-MM-DD HH:mm:ss'),
  dateEnd: moment().add(1, 'hour').format('YYYY-MM-DD HH:mm:ss'),
  option: '5 min',
  options: [
    { value: '5 minute', label: '5 min' },
    { value: '10 minute', label: '10 min' },
    { value: '15 minute', label: '15 min' },
    { value: '30 minute', label: '30 min' },
    { value: '1 hour', label: '1 hour' },
    { value: '2 hour', label: '2 hour' },
    { value: '1 day', label: '1 day' },
  ],
  momentFormat: 'YYYY-MM-DD HH:mm:ss',
};

export default filterWithDateRangeAndSelect;
