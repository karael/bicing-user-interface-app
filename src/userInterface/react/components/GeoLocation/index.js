import GeoLocator from 'userInterface/react/components/GeoLocation/GeoLocator';
import GeoSuggestion from 'userInterface/react/components/GeoLocation/GeoSuggestion';

export {
  GeoLocator,
  GeoSuggestion,
};
