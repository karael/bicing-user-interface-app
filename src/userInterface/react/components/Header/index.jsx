import React from 'react';

const Header = () => (
  <header className="App-header">
    <h1 className="App-title">Bicing!</h1>
  </header>);

export default Header;
