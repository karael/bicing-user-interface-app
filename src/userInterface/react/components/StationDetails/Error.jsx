import React from 'react';

const Error = () => (
  <div>
        StationDetails Error ....
  </div>
);

export default Error;
