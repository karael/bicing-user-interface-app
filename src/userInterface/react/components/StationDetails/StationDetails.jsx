/* eslint-disable */
import { shape, string } from 'prop-types';
import React from 'react';

const StationDetails = ({
  station: {
    name, type, address, addressNumber, zipCode,
  },
}) => (
  <div>
    <h6>{name}</h6>
    <p>
      {type}
      {address}
      {addressNumber}
      {zipCode}
    </p>
  </div>
);

StationDetails.propTypes = {
  station: shape({
    name: string.isRequired,
    type: string.isRequired,
    address: string,
    addressNumber: string,
    zipCode: string,
  }),
};

StationDetails.defaultProps = {
  address: '',
  addressNumber: '',
  zipCode: '',
};

export default StationDetails;
