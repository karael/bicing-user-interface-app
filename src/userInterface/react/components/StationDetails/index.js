import React from 'react';

import stationType from 'domain/types/stationType';

const StationDetails = ({ station }) => {
    return <div>
        <h6>{station.name}</h6>
        <p>
            {station.type}
            {station.address}
            {station.addressNumber}
            {station.zipCode}
            {/* {station.lastAvailability} */}
        </p>
    </div>
};

StationDetails.propTypes = {
    station: stationType.isRequired,
};

export default StationDetails;
