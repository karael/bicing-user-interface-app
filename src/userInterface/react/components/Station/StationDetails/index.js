import Error from 'userInterface/react/components/Station/StationDetails/Error';
import Loading from 'userInterface/react/components/Station/StationDetails/Loading';
import StationDetails from 'userInterface/react/components/Station/StationDetails/StationDetails';

export {
  Error,
  Loading,
  StationDetails,
};
