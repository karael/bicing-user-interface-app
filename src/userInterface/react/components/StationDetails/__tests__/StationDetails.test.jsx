import React from 'react';
import { create } from 'react-test-renderer';
import { StationDetails } from 'userInterface/react/components/StationDetails';

describe('userInterface/react/components/FilterWithDateRangeAndSelect/index.js', () => {
  test('it can renders with default values as expected', () => {
    const station = { name: '97 - C/ TARRAGONA 141', type: 'BIKE' };
    const component = create(<StationDetails station={station} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
  test('it can renders with values as expected', () => {
    const station = {
      name: '97 - C/ TARRAGONA 141',
      type: 'BIKE',
      address: 'Tarragona',
      addressNumber: '141',
      zipCode: '080014',
    };
    const component = create(<StationDetails station={station} />);


    expect(component.toJSON()).toMatchSnapshot();
  });
});
