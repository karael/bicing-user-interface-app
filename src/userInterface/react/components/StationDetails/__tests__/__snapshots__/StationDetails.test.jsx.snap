// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userInterface/react/components/FilterWithDateRangeAndSelect/index.js it can renders with default values as expected 1`] = `
<div>
  <h6>
    97 - C/ TARRAGONA 141
  </h6>
  <p>
    BIKE
  </p>
</div>
`;

exports[`userInterface/react/components/FilterWithDateRangeAndSelect/index.js it can renders with values as expected 1`] = `
<div>
  <h6>
    97 - C/ TARRAGONA 141
  </h6>
  <p>
    BIKE
    Tarragona
    141
    080014
  </p>
</div>
`;
